Lib/traceback.py

```diff
@@ -487,9 +487,11 @@ def __init__(self, exc_type, exc_value, exc_traceback, *, limit=None,
             _seen = set()
         _seen.add(id(exc_value))
 
-        # TODO: locals.
+        tb_gen = walk_tb(exc_traceback)
+        if frame_filter is not None:
+            tb_gen = filter(lambda p: frame_filter(p[0]), tb_gen)
```

Comment:
Note that I still needed a wrapper because the generator gives (frame, lineno) pairs.  I don't know if we want to user to have to accept the lineno in the filter as well.